ghost/members-api/lib/repositories/product.js

```diff
@@ -87,45 +100,47 @@ class ProductRepository {
                 });
             });
         }
+
+        let product = null;
+
         if ('stripe_product_id' in data) {
             const stripeProduct = await this._StripeProduct.findOne({
                 stripe_product_id: data.stripe_product_id
             }, options);
 
             if (!stripeProduct) {
-                return null;
+                product = null;
+            } else {
+                product = await stripeProduct.related('product').fetch(options);
             }
-
-            return await stripeProduct.related('product').fetch(options);
-        }
-
-        if ('stripe_price_id' in data) {
+        } else if ('stripe_price_id' in data) {
             const stripePrice = await this._StripePrice.findOne({
                 stripe_price_id: data.stripe_price_id
             }, options);
 
             if (!stripePrice) {
-                return null;
-            }
+                product = null;
+            } else {
+                const stripeProduct = await stripePrice.related('stripeProduct').fetch(options);
 
-            const stripeProduct = await stripePrice.related('stripeProduct').fetch(options);
+                if (!stripeProduct) {
+                    product = null;
+                }
 
-            if (!stripeProduct) {
-                return null;
+                product = await stripeProduct.related('product').fetch(options);
             }
-
-            return await stripeProduct.related('product').fetch(options);
-        }
-
-        if ('id' in data) {
-            return await this._Product.findOne({id: data.id}, options);
+        } else if ('id' in data) {
+            product = await this._Product.findOne({id: data.id}, options);
+        } else if ('slug' in data) {
+            product = await this._Product.findOne({slug: data.slug}, options);
         }
 
-        if ('slug' in data) {
-            return await this._Product.findOne({slug: data.slug}, options);
+        if (!product) {
+            throw new NotFoundError({message: 'Missing id, slug, stripe_product_id or stripe_price_id from data'});
```

Comment:
We never used to throw if it was missing, we'd return null

Reply:
oh!! good catch

Reply:
that was it :D thanks for saving loads of headache haha

Reply:
💃